### PR TITLE
test(qc): cover shared/ml_utils.py ML train/eval/persist helpers (0->27)

### PR DESCRIPTION
**Pull request by jsboige**
## What

Test coverage pour **`QuantConnect/shared/ml_utils.py`** (417L) — helper ML training/eval/persistence (`train_classifier`, `train_regressor`, `evaluate_model`, `save/load_to_objectstore`, `cross_validate_walk_forward`), consommé par notebooks 19-27 (ML/DL/AI). **0 test Python dédié** avant cette PR. Suite de la veine test-coverage QuantConnect/shared (suit `backtest_helpers.py` #7400 MERGED, `indicators.py` #7403, et la PR parallèle `features.py` #7407 de po-2026).

## Bug-hunt G.9 — 0 bug forçable, 2 quirks pinnés

Bug-hunt firsthand sur les 6 fonctions publiques : **0 bug fonctionnel à forcer-fix**. Math sklearn standard, guards `ValueError` (unknown model_type/task) et `FileNotFoundError` (missing ObjectStore key / local file) corrects, branches ObjectStore/local/ValueError défendables.

**2 quirks pinnés** (PAS bug-claims — module d'enseignement, one-subject-per-PR) :
1. **`evaluate_model` bare `except:` (L199)** : le bloc Sharpe optionnel attrape TOUT (y compris `KeyboardInterrupt`/`SystemExit`) — anti-pattern PEP8, mais défensif (le Sharpe est marqué "si applicable"). Pinné : `sharpe_ratio` est **optionnellement** présent (dépend de `predict_proba` + `std>0`), et fini quand présent.
2. **`cross_validate_walk_forward` default ratio dead-path** : `train_size_ratio=0.7` default → `train_initial=0.7N`, `test_size=0.3N`, 2e fold `train_end = 0.7N + 0.3N = N` → `test_end > total` → break. **Seul 1 fold s'exécute** quel que soit `n_splits` demandé (même expanding-window dead-path que `features.py walk_forward_split` signalé par po-2026 #7407). Configurable via ratio (`0.5` → plusieurs splits). Pinné dans `test_cross_validate_walk_forward_default_ratio_completes_one_split`.

## Livrable

- **`shared/test_ml_utils.py`** (co-localisé avec le module) — **27 tests CPU-only**, 1.74s, datasets **déterministes** (pas de RNG : `np.arange % k` + label parité, relationship linéaire pour regression) + un fake `qc_algorithm` dont `.ObjectStore` enregistre Save/ReadBytes/ContainsKey. Charge via `importlib` path-load (self-contained). Couvre :
  - **`train_classifier`** (7) : random_forest + logistic fit, kwargs override defaults, `random_state=42` default, unknown-type `ValueError`, **xgboost + lightgbm `ImportError` guards exercés directement** (pas installés localement → guard se déclenche naturellement, pas de mock).
  - **`train_regressor`** (4) : random_forest fit, **linear near-perfect** (y=2x₀+1 → `np.allclose(atol=1.0)`), svr fit, unknown-type `ValueError`.
  - **`evaluate_model`** (5) : classification keys + `accuracy∈[0,1]`, **perfect-model accuracy=1.0** pin, regression keys + **`rmse==sqrt(mse)`** invariant, unknown-task `ValueError`, Sharpe optionnel fini quand présent.
  - **`save/load` local** (4) : roundtrip pickle (predictions identiques), parent-dir creation, missing-file `FileNotFoundError`, no-target `ValueError`.
  - **`save/load` ObjectStore** (4) : roundtrip via fake qc, return `ObjectStore:<key>`, missing-key `FileNotFoundError`, **pickle-bytes verification** (`pickle.loads(raw)`), no-target `ValueError`.
  - **`cross_validate_walk_forward`** (3) : keys attendues, small-ratio multi-split complet, **default-ratio single-split quirk pinné**.

## Validation reproductible

```
$ cd MyIA.AI.Notebooks/QuantConnect/shared
$ python -m pytest test_ml_utils.py -q
27 passed in 1.74s
```

## Conformité

- Atomique : 1 domaine (QuantConnect/shared ML quality), 1 fichier test co-localisé.
- Anti-regression : nouveau fichier de test, **0 .py métier touché**, 0 changement de comportement (diff = 1 NEW file).
- Pas de `Closes #N` (contribution qualité standalone).
- Pas de C.2 implication : 0 notebook modifié (test-only). Les notebooks consumers (19-27) ne sont pas touchés.
- Aucune dépendance réseau, aucun QC Cloud, aucun ObjectStore réel (fake qc_algorithm), aucune clé API.
- Catalogue byte-identique à `main`.
- **Pre-flight collision check rigoureux** (leçon po-2026 c.537 + po-2024 c.694) : `gh pr list --state all --search "ml_utils"` = 2 résultats (#7407 features.py **fichier distinct**, #209 docs merged ancien) + file-set inspection de tous les QC/shared PRs ouverts (#7400 backtest_helpers, #7403 indicators, #7407 features, #7409 forge_helper) = **0 touche ml_utils.py**.
- Rebase frais off `origin/main` `c76d8b8f2`, worktree isolé `CoursIA-qc-mlutils`.

### CODE REVIEW

**Review by jsboige — COMMENTED** [review]
[Hermes] **COMMENT** — test-only PR, 0 LOC production touchée.

**Checklist exécutée :**
- ✅ Head SHA `017df5f` — 0 review/comment, pas de dédup
- ✅ Security scan : CLEAN (0 match)
- ✅ Scope : 1 fichier `test_ml_utils.py` ADDED (+345 / -0), module prod `shared/ml_utils.py` non modifié
- ✅ Hermeticité : datasets déterministes (`np.arange + modulo`, **no RNG**), `random_state=42` pin, `tmp_path`, `_FakeObjectStore` — pas de réseau ni QC Cloud

**Vérification non-superficielle :** les 27 tests couvrent train_classifier (5 types incl. xgboost/lightgbm `ImportError` guards), train_regressor, evaluate_model (métriques + Sharpe optionnel), save/load roundtrips (local + ObjectStore + pickle-bytes check), cross_validate_walk_forward. Le `test_cross_validate_walk_forward_default_ratio_completes_one_split` **pigne un quirk documenté** (n_splits=1 avec ratio 0.7) avec un commentaire explicite — bonne discipline (pin, pas bug-claim caché).

RAS. Coverage réelle (0 → 27), pas juste du adding-tests-for-the-sake-of.
